pkgs/misc/uboot/hardkernel-firmware.nix

```diff
@@ -0,0 +1,126 @@
+
+let
+  buildHardkernelFirmware = {
+    version ? null
+    , src ? null
+    , name ? ""
+    , filesToInstall
+    , installDir ? "$out"
+    , defconfig
+    , extraMeta ? {}
+    , ... } @ args: (overrideCC stdenv buildPackages.gcc49).mkDerivation ({
+      pname = "uboot-hardkernel-firmware-${name}";
+
+      nativeBuildInputs = [
+        buildPackages.git
+        buildPackages.hostname
+        pkgsCross.arm-embedded.stdenv.cc
+      ];
+      depsBuildBuild = [ buildPackages.stdenv.cc ];
+
+      postPatch = ''
+        substituteInPlace Makefile --replace "/bin/pwd" "pwd"
+      '';
+
+      makeFlags = [
+        "CROSS_COMPILE=${stdenv.cc.targetPrefix}"
+        "CROSS_COMPILE_32=${pkgsCross.arm-embedded.stdenv.cc.targetPrefix}"
+        "${defconfig}" "bl301.bin"
+      ];
+
+      installPhase = ''
+        mkdir -p ${installDir}
+        cp ${lib.concatStringsSep " " filesToInstall} ${installDir}
+      '';
+
+      meta = with lib; {
+        homepage = "https://www.hardkernel.com/";
+        description = "Das U-Boot from Hardkernel with Odroid embedded devices firmware and support";
+        license = licenses.unfreeRedistributableFirmware;
+        maintainers = with maintainers; [ aarapov ];
+      } // extraMeta;
+    } // removeAttrs args [ "extraMeta" ]);
+in {
+  inherit buildHardkernelFirmware;
+
+  # https://wiki.odroid.com/odroid-c2/software/building_u-boot
+  # $ nix-build -A pkgsCross.aarch64-multiplatform.firmwareOdroidC2
+  firmwareOdroidC2 = let
+    name = "odroidc2";
+    odroidc2-bl301 = fetchgit {
+      url = "https://github.com/hardkernel/u-boot_firmware.git";
+      rev = "b7b90c1099b057d35ebae886b7846b5d9bfb4143"; # "odroidc2-bl301"
+      sha256 = "0kdb1mg5zd7qyabfpbh98cs07icfzpkywvva13ybf4mf5g50g0n3";
+      deepClone = true;
+    };
+  in buildHardkernelFirmware {
+    defconfig = "odroidc2_config";
+    name = "firmware-odroid-c2";
+    version = "2015.01";
+    src = fetchgit {
+      url = "https://github.com/hardkernel/u-boot.git";
+      rev = "fac4d2da0a1b61dfdeaca0034a45151ff5983fb8"; # "odroidc2-v2015.01"
+      sha256 = "1wslpg7nrbzqns1gglkxgi1ka2rpmpmhigk4p4f4589lkky4jrmd";
+      leaveDotGit = true;
+    };
+
+    prePatch = ''
+      git remote add u-boot_firmware ${odroidc2-bl301}/
+      git fetch u-boot_firmware
+      git cherry-pick --no-commit \
+        5ce504067bb83de03d17173d5585e849df5d5a33^..${odroidc2-bl301.rev}
+
+      substituteInPlace ./arch/arm/cpu/armv8/gxb/firmware/scp_task/Makefile \
+        --replace "CROSS_COMPILE" "CROSS_COMPILE_32"
+    '';
+
+    postBuild = ''
+      patchelf --set-interpreter "$(cat ${buildPackages.stdenv.cc}/nix-support/dynamic-linker)" fip/fip_create
+    '';
+
+    filesToInstall = [
+      "build/scp_task/bl301.bin" "fip/fip_create" "fip/gxb/aml_encrypt_gxb"
+      "fip/gxb/bl30.bin" "fip/gxb/bl31.bin" "fip/gxb/bl2.package"
+      "sd_fuse/sd_fusing.sh" "sd_fuse/bl1.bin.hardkernel"
+    ];
+
+    extraMeta.platforms = [ "aarch64-linux" ];
```

Comment:
For this to be accurate, we need to not install the x86_64 binaries (which aren't needed as we have open source alternatives).

Reply:
documented this case in code and made builds gracefull

Reply:
Fixed, this derivative does not produce x86_64 binaries.